### ISSUE TITLE
VSCode Codeium extension returns error :"cannot find language server port file". Then I tried to download its windows version from here, sync the version of extension and language_server to 1.2.97, but still doesn't work.

### ISSUE THREAD

**Issue by JasonJarvan** [Exafunction/codeium#13]
The background is that Codeium worked well in my VSCode before. Just from today, 
I meet the problem that "cannot find language server port file in C:\Users\xxx\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager in 60 secs"
I am not sure whether this is due to my Internet problem because I am in Hongkong, so I found a way that I can download the corresponding version of language server from this repo. So I download its windows version from here, (I am working in Windows OS) and sync the version of extension and language_server to 1.2.97.(I am using the 1.2.97 version extension)
Then I found the temp file folder of Codeium always change when I restart my VSCode, so I check the log in VSCode and found the temp folder, and put language server file into it. The full path is "C:\Users\xxx\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager\language_server_windows_x64.exe"
Then the error "cannot find language server port file" doesnt appear, but codeium still doesn't work and return the following message:

The log after I put server file is:
```
2023-10-07 18:10:43.402 [INFO]: win-ca initialized
2023-10-07 18:10:43.404 [ERROR]: undefined
2023-10-07 18:10:46.558 [INFO]: Creating manager directory: C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager
2023-10-07 18:10:49.507 [INFO]: Using cached LS with valid crc32: c:\Users\Sheng.Zhao\.vscode\extensions\codeium.codeium-1.2.97\dist\f3bda5b3eb4dc7c7904cb8e9a3156e9f25f1ebb9\language_server_windows_x64.exe
2023-10-07 18:10:49.520 [INFO]: Language server started
2023-10-07 18:10:50.581 [INFO]: [Language Server (stderr)]:
I1007 18:10:50.580641 18748 main.go:452] Starting language server manager with pid 18748
2023-10-07 18:10:50.590 [INFO]: [Language Server (stderr)]:
I1007 18:10:50.590636 18748 main.go:106] Manager acquired its lock file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/locks/manager.lock
2023-10-07 18:10:50.656 [INFO]: [Language Server (stderr)]:
I1007 18:10:50.622654 18748 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/child_random_port_1696673450590636400_4784791657710900675 to be created
2023-10-07 18:10:51.159 [INFO]: [Language Server (stderr)]:
I1007 18:10:51.142084 38048 main.go:446] Starting language server process with pid 38048
2023-10-07 18:10:52.375 [INFO]: [Language Server (stderr)]:
I1007 18:10:52.375863 38048 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:10:52.410 [INFO]: [Language Server (stderr)]:
2023/10/07 18:10:52 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:10:52.609 [INFO]: [Language Server (stderr)]:
I1007 18:10:52.610101 38048 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:10:52.610 [INFO]: [Language Server (stderr)]:
I1007 18:10:52.610101 38048 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:10:52.815 [INFO]: [Language Server (stderr)]:
I1007 18:10:52.801078 38048 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:10:53.349 [INFO]: [Language Server (stderr)]:
I1007 18:10:53.344298 38048 api_server_client.go:115] Successfully dialed proxy
I1007 18:10:53.344298 38048 server.go:387] Successfully created API server client
I1007 18:10:53.344298 38048 api_server_client.go:115] Successfully dialed proxy
I1007 18:10:53.344298 38048 server.go:393] Successfully created API streaming client
2023-10-07 18:10:53.400 [INFO]: [Language Server (stderr)]:
I1007 18:10:53.396303 38048 server.go:402] Successfully initialized tokenizer
I1007 18:10:53.396303 38048 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:10:59.634 [INFO]: Found language server running on port NaN
2023-10-07 18:11:05.645 [INFO]: [Language Server (stderr)]:
I1007 18:11:05.646115 18748 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:11:09.692 [INFO]: [Language Server (stderr)]:
E1007 18:11:09.692282 18748 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 38048 at 1007 18:11:09.691798, sent SIGTERM then SIGKILL
2023-10-07 18:11:09.693 [INFO]: [Language Server (stderr)]:
E1007 18:11:09.692946 18748 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673450597636600_2497693638325532256: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673450597636600_2497693638325532256: The process cannot access the file because it is being used by another process.
2023-10-07 18:11:09.709 [INFO]: [Language Server (stderr)]:
I1007 18:11:09.710129 18748 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/child_random_port_1696673469695244700_7736025092012760781 to be created
2023-10-07 18:11:09.791 [INFO]: [Language Server (stderr)]:
I1007 18:11:09.790375  9616 main.go:446] Starting language server process with pid 9616
2023-10-07 18:11:09.868 [INFO]: [Language Server (stderr)]:
I1007 18:11:09.869222  9616 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:11:09.881 [INFO]: [Language Server (stderr)]:
2023/10/07 18:11:09 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:11:09.884 [INFO]: [Language Server (stderr)]:
I1007 18:11:09.885304  9616 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:11:09.885 [INFO]: [Language Server (stderr)]:
I1007 18:11:09.885854  9616 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:11:10.058 [INFO]: [Language Server (stderr)]:
I1007 18:11:10.059575  9616 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:11:10.455 [INFO]: [Language Server (stderr)]:
I1007 18:11:10.457069  9616 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:11:10.456 [INFO]: [Language Server (stderr)]:
I1007 18:11:10.457069  9616 server.go:387] Successfully created API server client
I1007 18:11:10.457069  9616 api_server_client.go:115] Successfully dialed proxy
I1007 18:11:10.457637  9616 server.go:393] Successfully created API streaming client
2023-10-07 18:11:10.509 [INFO]: [Language Server (stderr)]:
I1007 18:11:10.510208  9616 server.go:402] Successfully initialized tokenizer
I1007 18:11:10.510208  9616 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:11:24.790 [INFO]: [Language Server (stderr)]:
I1007 18:11:24.791703 18748 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:11:28.829 [INFO]: [Language Server (stderr)]:
E1007 18:11:28.830910 18748 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 9616 at 1007 18:11:28.830372, sent SIGTERM then SIGKILL
2023-10-07 18:11:28.830 [INFO]: [Language Server (stderr)]:
E1007 18:11:28.830910 18748 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673469695996600_6161909910470624704: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673469695996600_6161909910470624704: The process cannot access the file because it is being used by another process.
2023-10-07 18:11:28.841 [INFO]: [Language Server (stderr)]:
I1007 18:11:28.843497 18748 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/child_random_port_1696673488830910500_4515010168728142365 to be created
2023-10-07 18:11:28.908 [INFO]: [Language Server (stderr)]:
I1007 18:11:28.908338 69588 main.go:446] Starting language server process with pid 69588
2023-10-07 18:11:28.980 [INFO]: [Language Server (stderr)]:
I1007 18:11:28.981018 69588 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:11:28.993 [INFO]: [Language Server (stderr)]:
2023/10/07 18:11:28 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:11:28.998 [INFO]: [Language Server (stderr)]:
I1007 18:11:28.998702 69588 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:11:28.998 [INFO]: [Language Server (stderr)]:
I1007 18:11:28.999711 69588 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:11:29.176 [INFO]: [Language Server (stderr)]:
I1007 18:11:29.177289 69588 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:11:29.552 [INFO]: [Language Server (stderr)]:
I1007 18:11:29.553331 69588 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:11:29.552 [INFO]: [Language Server (stderr)]:
I1007 18:11:29.553893 69588 server.go:387] Successfully created API server client
I1007 18:11:29.553893 69588 api_server_client.go:115] Successfully dialed proxy
I1007 18:11:29.554444 69588 server.go:393] Successfully created API streaming client
2023-10-07 18:11:29.603 [INFO]: [Language Server (stderr)]:
I1007 18:11:29.603926 69588 server.go:402] Successfully initialized tokenizer
I1007 18:11:29.604892 69588 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:11:43.888 [INFO]: [Language Server (stderr)]:
I1007 18:11:43.888174 18748 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:11:47.933 [INFO]: [Language Server (stderr)]:
E1007 18:11:47.933775 18748 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 69588 at 1007 18:11:47.933775, sent SIGTERM then SIGKILL
2023-10-07 18:11:47.934 [INFO]: [Language Server (stderr)]:
E1007 18:11:47.934287 18748 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673488832094600_7105776963146391532: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/server_outputs/language_server_stderr_1696673488832094600_7105776963146391532: The process cannot access the file because it is being used by another process.
2023-10-07 18:11:47.948 [INFO]: [Language Server (stderr)]:
I1007 18:11:47.949044 18748 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\manager/child_random_port_1696673507934404200_8713096811381230606 to be created
2023-10-07 18:11:48.022 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.021512  6400 main.go:446] Starting language server process with pid 6400
2023-10-07 18:11:48.088 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.087193  6400 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:11:48.099 [INFO]: [Language Server (stderr)]:
2023/10/07 18:11:48 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:11:48.101 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.102262  6400 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
I1007 18:11:48.102285  6400 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:11:48.288 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.288511  6400 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:11:48.663 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.663434  6400 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:11:48.663 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.663434  6400 server.go:387] Successfully created API server client
I1007 18:11:48.663434  6400 api_server_client.go:115] Successfully dialed proxy
I1007 18:11:48.663980  6400 server.go:393] Successfully created API streaming client
2023-10-07 18:11:48.711 [INFO]: [Language Server (stderr)]:
I1007 18:11:48.710520  6400 server.go:402] Successfully initialized tokenizer
I1007 18:11:48.711527  6400 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:11:52.961 [INFO]: [Language Server (stderr)]:
F1007 18:11:52.961575 18748 main.go:246] Failed to convert random server port to integer: strconv.Atoi: parsing "language": invalid syntax
2023-10-07 18:11:52.971 [INFO]: Language server exited with code 1
```

**Comment by JasonJarvan** [comment]
This time I put it in "C:\Users\xxx\AppData\Local\Temp\460a35fe-dc23-4428-85ba-58051a947d18\codeium\language_server_windows_x64.exe" then it seems language server started, but "Exceeded maximum number of connection failures 4"
```
2023-10-07 18:34:01.011 [INFO]: win-ca initialized
2023-10-07 18:34:01.013 [ERROR]: undefined
2023-10-07 18:34:02.572 [INFO]: Creating manager directory: C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager
2023-10-07 18:34:03.326 [INFO]: Using cached LS with valid crc32: c:\Users\Sheng.Zhao\.vscode\extensions\codeium.codeium-1.2.97\dist\f3bda5b3eb4dc7c7904cb8e9a3156e9f25f1ebb9\language_server_windows_x64.exe
2023-10-07 18:34:03.338 [INFO]: Language server started
2023-10-07 18:34:03.852 [INFO]: [Language Server (stderr)]:
I1007 18:34:03.829574 71920 main.go:452] Starting language server manager with pid 71920
I1007 18:34:03.834315 71920 main.go:106] Manager acquired its lock file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/locks/manager.lock
2023-10-07 18:34:03.872 [INFO]: [Language Server (stderr)]:
I1007 18:34:03.871912 71920 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/child_random_port_1696674843834315200_4125631535359814054 to be created
2023-10-07 18:34:07.323 [INFO]: [Language Server (stderr)]:
I1007 18:34:07.250398  4828 main.go:446] Starting language server process with pid 4828
2023-10-07 18:34:08.549 [INFO]: [Language Server (stderr)]:
I1007 18:34:08.541933  4828 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:34:08.699 [INFO]: [Language Server (stderr)]:
2023/10/07 18:34:08 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:34:08.700 [INFO]: [Language Server (stderr)]:
I1007 18:34:08.695197  4828 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
I1007 18:34:08.695197  4828 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:34:08.925 [INFO]: [Language Server (stderr)]:
I1007 18:34:08.913665  4828 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:34:09.411 [INFO]: [Language Server (stderr)]:
I1007 18:34:09.390824  4828 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:34:09.414 [INFO]: [Language Server (stderr)]:
I1007 18:34:09.391825  4828 server.go:387] Successfully created API server client
I1007 18:34:09.391825  4828 api_server_client.go:115] Successfully dialed proxy
I1007 18:34:09.392822  4828 server.go:393] Successfully created API streaming client
2023-10-07 18:34:09.556 [INFO]: [Language Server (stderr)]:
I1007 18:34:09.555920  4828 server.go:402] Successfully initialized tokenizer
2023-10-07 18:34:09.561 [INFO]: [Language Server (stderr)]:
I1007 18:34:09.555920  4828 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:34:18.922 [INFO]: [Language Server (stderr)]:
I1007 18:34:18.922164 71920 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:34:22.970 [INFO]: [Language Server (stderr)]:
E1007 18:34:22.970858 71920 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 4828 at 1007 18:34:22.970858, sent SIGTERM then SIGKILL
2023-10-07 18:34:22.971 [INFO]: [Language Server (stderr)]:
E1007 18:34:22.971411 71920 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674843835157900_1286647082089584530: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674843835157900_1286647082089584530: The process cannot access the file because it is being used by another process.
2023-10-07 18:34:22.992 [INFO]: [Language Server (stderr)]:
I1007 18:34:22.993345 71920 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/child_random_port_1696674862971411500_4058803318937636987 to be created
2023-10-07 18:34:23.072 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.071946 61604 main.go:446] Starting language server process with pid 61604
2023-10-07 18:34:23.165 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.166690 61604 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:34:23.181 [INFO]: [Language Server (stderr)]:
2023/10/07 18:34:23 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:34:23.185 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.186086 61604 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:34:23.186 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.186086 61604 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:34:23.365 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.365373 61604 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:34:23.777 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.777844 61604 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:34:23.778 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.778375 61604 server.go:387] Successfully created API server client
I1007 18:34:23.778375 61604 api_server_client.go:115] Successfully dialed proxy
I1007 18:34:23.778375 61604 server.go:393] Successfully created API streaming client
2023-10-07 18:34:23.840 [INFO]: [Language Server (stderr)]:
I1007 18:34:23.840859 61604 server.go:402] Successfully initialized tokenizer
I1007 18:34:23.841314 61604 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:34:33.980 [INFO]: Found language server running on port NaN
2023-10-07 18:34:38.009 [INFO]: [Language Server (stderr)]:
I1007 18:34:38.010158 71920 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:34:42.057 [INFO]: [Language Server (stderr)]:
E1007 18:34:42.050457 71920 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 61604 at 1007 18:34:42.050457, sent SIGTERM then SIGKILL
E1007 18:34:42.051005 71920 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674862977049700_4382251691431940731: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674862977049700_4382251691431940731: The process cannot access the file because it is being used by another process.
2023-10-07 18:34:42.076 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.070887 71920 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/child_random_port_1696674882051005000_6577768031221963204 to be created
2023-10-07 18:34:42.149 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.148847 71800 main.go:446] Starting language server process with pid 71800
2023-10-07 18:34:42.239 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.235767 71800 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:34:42.248 [INFO]: [Language Server (stderr)]:
2023/10/07 18:34:42 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:34:42.251 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.252129 71800 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:34:42.252 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.252129 71800 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:34:42.426 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.427574 71800 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:34:42.807 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.808389 71800 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:34:42.808 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.808899 71800 server.go:387] Successfully created API server client
I1007 18:34:42.808941 71800 api_server_client.go:115] Successfully dialed proxy
I1007 18:34:42.808941 71800 server.go:393] Successfully created API streaming client
2023-10-07 18:34:42.857 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.857583 71800 server.go:402] Successfully initialized tokenizer
2023-10-07 18:34:42.858 [INFO]: [Language Server (stderr)]:
I1007 18:34:42.857583 71800 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:34:57.097 [INFO]: [Language Server (stderr)]:
I1007 18:34:57.096001 71920 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:35:01.143 [INFO]: [Language Server (stderr)]:
E1007 18:35:01.143245 71920 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 71800 at 1007 18:35:01.142720, sent SIGTERM then SIGKILL
2023-10-07 18:35:01.144 [INFO]: [Language Server (stderr)]:
E1007 18:35:01.144246 71920 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674882051666100_5769327222151780204: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674882051666100_5769327222151780204: The process cannot access the file because it is being used by another process.
2023-10-07 18:35:01.163 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.163447 71920 main.go:228] Waiting for random port file in C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/child_random_port_1696674901144782500_1762242648314721118 to be created
2023-10-07 18:35:01.237 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.235309 62304 main.go:446] Starting language server process with pid 62304
2023-10-07 18:35:01.316 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.314575 62304 proxy.go:93] proxyplease.proxy> No proxy provided. Attempting to infer from system.
2023-10-07 18:35:01.335 [INFO]: [Language Server (stderr)]:
2023/10/07 18:35:01 [proxy.Provider.readWinHttpProxy] No proxy discovered via AutoDetect: winapi error #12180
2023-10-07 18:35:01.341 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.338131 62304 proxy.go:93] proxyplease.proxy> Inferred proxy from system: http://192.168.104.3:8080
2023-10-07 18:35:01.342 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.338131 62304 client.go:589] [DEBUG] GET https://server.codeium.com/exa.api_server_pb.ApiServerService/GetCompletions
2023-10-07 18:35:01.516 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.516088 62304 proxy.go:94] proxyplease.connect> Proxy successfully established. No authentication was required.
2023-10-07 18:35:01.891 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.890635 62304 api_server_client.go:115] Successfully dialed proxy
2023-10-07 18:35:01.892 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.891185 62304 server.go:387] Successfully created API server client
I1007 18:35:01.891185 62304 api_server_client.go:115] Successfully dialed proxy
I1007 18:35:01.891733 62304 server.go:393] Successfully created API streaming client
2023-10-07 18:35:01.945 [INFO]: [Language Server (stderr)]:
I1007 18:35:01.944027 62304 server.go:402] Successfully initialized tokenizer
I1007 18:35:01.944569 62304 unleash.go:57] Initializing Unleash with production environment
2023-10-07 18:35:16.199 [INFO]: [Language Server (stderr)]:
I1007 18:35:16.198811 71920 main.go:259] Failed to get random ports within 15 seconds
2023-10-07 18:35:20.238 [INFO]: [Language Server (stderr)]:
E1007 18:35:20.238764 71920 cmd_wrapper.go:266] LanguageServer wait failed: timed out while waiting on process 62304 at 1007 18:35:20.238213, sent SIGTERM then SIGKILL
2023-10-07 18:35:20.239 [INFO]: [Language Server (stderr)]:
E1007 18:35:20.238764 71920 main.go:211] Failed to remove stderr file C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674901146312900_7595255559110476309: remove C:\Users\SHENG~1.ZHA\AppData\Local\Temp\db410970-e0fd-4b26-a96b-2328adaaab48\codeium\manager/server_outputs/language_server_stderr_1696674901146312900_7595255559110476309: The process cannot access the file because it is being used by another process.
F1007 18:35:20.238764 71920 main.go:152] Exceeded maximum number of connection failures 4
2023-10-07 18:35:20.245 [INFO]: Language server exited with code 1
```

**Comment by Ottepel84** [comment]
Th same problem. Works only when installing version 1.4.27. When trying to install 1.6.10 or 1.6.11 it does not work and displays a message "cannot find language server port file".

**Comment by pqn** [comment]
We are no longer supporting before Windows 10 in newer releases, basically anything 1.6.9 or above. [VS Code already has this requirement.](https://code.visualstudio.com/docs/supporting/requirements) You can stay pinned on an older release indefinitely though.

**Comment by pqn** [comment]
My comment is a response to @Ottepel84's post, but closing the issue anyway due to staleness (feel free to open a new issue if this is reproducible with the latest release).
